cli/src/commit_templater.rs

```diff
@@ -48,8 +52,27 @@ pub struct CommitTemplateLanguage<'repo> {
     keyword_cache: CommitKeywordCache,
 }
 
-pub trait CommitTemplateLanguageExtension {
-    fn build_fn_table<'repo>(&self) -> CommitTemplateBuildFnTable<'repo>;
+impl<'repo> CommitTemplateLanguage<'repo> {
+    /// Sets up environment where commit template will be transformed to
+    /// evaluation tree.
+    pub fn new(
+        repo: &'repo dyn Repo,
+        workspace_id: &WorkspaceId,
```

Comment:
Take by value?  Similarly in the other constructor

Reply:
Could be.

I thought they could be `Into<WorkspaceId>` for convenience, but apparently our id types don't implement reference-to-owned conversion (maybe because there are no slice types yet.)

All callers usually pass ids by reference, so I'm not sure which is better right now.

Reply:
I'll go with pass-by-ref because these ids doesn't have to be owned type, conceptually. I just made them cloned because lifetime notation was noisy and clone should be cheap.